utils/funcexp/functor.h

```diff
@@ -38,6 +38,8 @@
 
 #include "dataconvert.h"
 
+#include <llvm/IR/IRBuilder.h>
```

Comment:
This is a rather large header file that will `#include` many other header files. I think that including that kind of a "jungle" in a rather popular header file (`functor.h` is being included from many other header files, which in turn could be included from many compilation units) could significantly include the compilation time.

We only seem to need this for the type name `llvm::Value`, which `compile()` would return a pointer to. I think that it would be better to define an opaque type and avoid including the header here. Something like this might work:
```c++
namespace llvm { class Value; }
```